api/v1alpha5/metal3machine_types.go

```diff
@@ -62,6 +62,12 @@ type Metal3MachineSpec struct {
 	// NetworkData is an object storing the reference to the secret containing the
 	// network data given by the user.
 	NetworkData *corev1.SecretReference `json:"networkData,omitempty"`
+
+	// When set to disabled, automated cleaning will be skipped
+	// during provisioning and deprovisioning.
+	// +kubebuilder:default:=metadata
+	// +kubebuilder:validation:Enum:=metadata;disabled
+	AutomatedCleaningMode string `json:"automatedCleaningMode,omitempty"`
```

Comment:
Field name is not descriptive enough, `AutomatedCleaningMode`
What is it cleaning ? disk ?
Which disk is it cleaning ? non-boot disk ?

Reply:
It is about secondary storage (i.e. hosted storage). I think having a good documentation will answer those questions.